src/test/kotlin/org/jetbrains/kotlinx/jupyter/test/repl/ReplTests.kt

```diff
@@ -439,3 +441,335 @@ class ReplTests : AbstractSingleReplTest() {
         assertEquals("org.RDKit.RWMol", res!!::class.qualifiedName)
     }
 }
+
+class ReplVarsTest : AbstractSingleReplTest() {
+    override val repl = makeSimpleRepl()
+
+    @Test
+    fun testVarsStateConsistency() {
+        assertTrue(repl.notebook.variablesState.isEmpty())
+        val res = eval(
+            """
+            val x = 1 
+            val y = 0
+            val z = 47
+            """.trimIndent()
+        )
+
+        val varsUpdate = mutableMapOf<String, String>(
+            "x" to "1",
+            "y" to "0",
+            "z" to "47"
+        )
+        assertEquals(res.metadata.evaluatedVariablesState, varsUpdate)
+        assertFalse(repl.notebook.variablesState.isEmpty())
+        val varsState = repl.notebook.variablesState
+        assertEquals("1", varsState["x"]!!.stringValue)
```

Comment:
Let's add extension method that will allow us to write `varsState.getStringValue("x")`. And also, `varsState.getValue("x")` for getting raw value